sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/AadOAuth2ClientAuthenticationJwkResolver.java

```diff
@@ -0,0 +1,95 @@
+import java.util.UUID;
+
+/**
+ * An {@link RSAKey} resolver implementation parses the certificate locally.
+ *
+ * @since 4.3.0
+ * @see <a href="https://docs.microsoft.com/azure/active-directory/develop/active-directory-certificate-credentials">Certificate credentials</a>
+ */
+public class AadOAuth2ClientAuthenticationJwkResolver implements OAuth2ClientAuthenticationJwkResolver {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AadOAuth2ClientAuthenticationJwkResolver.class);
+
+    private final String clientCertificatePath;
+    private final String clientCertificatePassword;
+
+    /**
+     * Creates a new instance of {@link AadOAuth2ClientAuthenticationJwkResolver}
+     * @param clientCertificatePath the client certificate path
+     * @param clientCertificatePassword the client certificate password
+     */
+    public AadOAuth2ClientAuthenticationJwkResolver(String clientCertificatePath,
+                                                    String clientCertificatePassword) {
+        Assert.notNull(clientCertificatePath, "clientCertificatePath cannot be null");
+        Assert.notNull(clientCertificatePassword, "clientCertificatePassword cannot be null");
+
+        String fileExtension = clientCertificatePath.substring(clientCertificatePath.lastIndexOf(".") + 1);
+        Assert.isTrue("pfx".equals(fileExtension) || "p12".equals(fileExtension),
+            "Only files with the '.pfx' or '.p12' extension are supported.");
+
+        this.clientCertificatePath = clientCertificatePath;
+        this.clientCertificatePassword = clientCertificatePassword;
+    }
+
+    @Override
+    public JWK resolve(ClientRegistration clientRegistration) {
+        if (ClientAuthenticationMethod.PRIVATE_KEY_JWT.equals(clientRegistration.getClientAuthenticationMethod())) {
+            try (FileInputStream inputStream = new FileInputStream(clientCertificatePath)) {
+                KeyStore keyStore = KeyStore.getInstance("PKCS12");
+                char[] password = clientCertificatePassword.toCharArray();
+                keyStore.load(inputStream, password);
+                String alias = keyStore.aliases().nextElement();
+                PrivateKey privateKey = (PrivateKey) keyStore.getKey(alias, password);
+                X509Certificate x509Certificate = (X509Certificate) keyStore.getCertificate(alias);
+                PublicKey publicKey = x509Certificate.getPublicKey();
+                return new RSAKey.Builder((RSAPublicKey) publicKey)
+                    .privateKey(privateKey)
+                    .x509CertThumbprint(Base64URL.encode(getX5t(x509Certificate)))
+                    .keyID(UUID.randomUUID().toString())
+                    .build();
+            } catch (KeyStoreException | IOException | NoSuchAlgorithmException
+                     | CertificateException | UnrecoverableKeyException e) {
+                LOGGER.error("Resolve RSAKey exception.", e);
+            }
+        }
+        return null;
```

Comment:
should we return null?

Reply:
Currently, only supports the PRIVATE_KEY_JWT.